documentation/modules/operators/ref-operator-cluster.adoc

```diff
@@ -171,6 +171,51 @@ If you are using a different DNS domain name suffix in your cluster, change the
 The timeout for building new Kafka Connect images with additional connectots, in milliseconds.
 This value should be increased when using Strimzi to build container images containing many connectors or using a slow container registry.
 
+`STRIMZI_FEATURE_GATES`:: Optional.
+Disables or enables feature gates to activate new features or change the operator behavior.
+For more details about the feature gates, see xref:ref-operator-cluster-feature-gates-{context}[].
+
+[id='ref-operator-cluster-feature-gates-{context}']
+== Feature Gates
+
+Strimzi operators supports feature gates to gate selected functionality and switch it on or off.
+Each feature gate has a default state (enabled or disabled by default) and users are to manually enable or disable them.
+The feature gates always apply for the whole operator and alter its behavior.
+
+The feature gates should go through 3 different maturity stages:
+* Alpha (typically disabled by default)
+* Beta (typically enabled by default)
+* General Availability / GA (typically enabled by default or always enabled)
+
+Alpha stage indicates that the feature might be unstable (it might change in the future), experimental or not sufficiently tested for production use.
+Beta stage indicates a that the feature is already well tested and its functionality should most probably not change.
+GA means that the feature is stable and should not change anymore.
+GA features might optionally also change to always enabled features.
+In this case the feature gate will be removed, and it will not be possible to disable the feature anymore.
+The alpha and beta features might be removed in case they are deemed as not useful or not the solution.
+
+You can configure the feature gates using the environment variable `STRIMZI_FEATURE_GATES`.
+It should contain a comma separate list of the feature gate names prefixed with `+` or `-` for enabling or disabling the feature gate.
+
+.Example feature gate configuration which enables `FeatureGate1` and disables `FeatureGate2`.
+[source,yaml,options="nowrap"]
+----
+env:
+  - name: STRIMZI_FEATURE_GATES
+    value: +FeatureGate1,-FeatureGate2
+----
+
+Feature gates can be in 3 different phases indicating their maturity level:
+* _Alpha_ features
+
+=== Control Plane Listener
+
+When enabled, the `ControlPlaneListener` feature gate will configure Kafka brokers to use a separate listener for controller connections and for data connections (replication).
+This feature gate is currently in the alpha phase and is disabled by default.
+To enabled it, use `+ControlPlaneListener` in the `STRIMZI_FEATURE_GATES` environment variable in the Cluster Operator configuration.
+This feature gate has to be disabled when upgrading from Strimzi 0.22 and earlier or when downgrading to Strimzi 0.22 and earlier.
+This feature is introduced in Strimzi 0.23 and is expected to remain in the alpha phase for four releases before moving to the beta phase and be enabled by default.
```

Comment:
```suggestion
The `ControlPlaneListener` feature gate was introduced in Strimzi 0.23 and is expected to remain in the alpha phase for four releases before it moves to the beta phase and is enabled by default.
```

Reply:
If we need to be more specific^^

Reply:
I wonder about a table for the feature gates, showing the version in which it was added (alpha), the version in which it became beta and the version in which is became GA. That would allow us to define the upgrade/downgrade steps generically and also give people a quick summary of the available gates. We could like to a more detailed description of each gate that was outside the table.

Reply:
I guess the question is whether we should add the table now or later. RIght now, with just one feature gate in one state, it would look weird:

| Feature Gate        | Alpha          | Beta  | GA|
| ------------- |:-------------:| -----:| -----:|
| `ControlPlaneListener`      | 0.23.0 |  |  |

Reply:
@PaulRMellor @tombentley Ok, so I added the table with the overview of the gates. Please have a look. Even with only one feature gate right now I think it looks good.

![Screenshot 2021-05-07 at 10 54 14](https://user-images.githubusercontent.com/5658439/117424956-9e07eb00-af22-11eb-8d3c-0526a6dc80b6.png)